packages/forma-36-react-components/src/components/IconButton/IconButton.tsx

```diff
@@ -5,6 +5,36 @@ import { Icon, IconProps } from '../Icon';
 import { TabFocusTrap } from '../TabFocusTrap';
 import styles from './IconButton.css';
 
+type IconButtonAnchorProps =
+  | {
+      /**
+       * Used to make the decision of either rendering as a <a> or as a <button> tag
+       */
+      href?: undefined;
+      /**
+       * Used with href to define a relationship between a linked resource and the current document
+       */
+      rel?: never;
+      /**
+       * Used with href to specify target attribute value
+       */
+      target?: never;
+    }
+  | {
+      /**
+       * Used to make the decision of either rendering as a <a> or as a <button> tag
+       */
+      href: string;
+      /**
+       * Used with href to define a relationship between a linked resource and the current document
+       */
+      rel?: React.AnchorHTMLAttributes<HTMLAnchorElement>['rel'];
+      /**
+       * Used with href to specify target attribute value
+       */
+      target?: React.AnchorHTMLAttributes<HTMLAnchorElement>['target'];
+    };
+
 export interface IconButtonProps extends React.HTMLAttributes<HTMLElement> {
```

Comment:
`export interface IconButtonProps extends React.HTMLAttributes<HTMLAnchorElement | HTMLButtonElement>`

I recommend extending the built-in types from typescript, so we don't have to redefine them

Reply:
sure thing :) is this what you had in mind?